client/src/components/editor/LinkSourceAnet.js

```diff
@@ -66,6 +66,7 @@ const LinkSourceAnet = ({
               const anetLinkNode = createAnetLinkNode(objectType, value.uuid)
               insertAnetLink(anetLinkNode)
             }}
+            keepSearchText
```

Comment:
Solution doesn't work for e.g. **Additional engagement needed for** (in report form, custom field under Engagement information) and in the related objects of a note: in both places it doesn't remember the search input when changing the entity type.
Additionally, a change like this could do with some client-side tests to verify that the current implementation fails the requirement, and that the new (intended) implementation meets it.

Reply:
Actually, the main intent was to support this functionality in rich text-editor. Therefore, of course it will not work for other components. After adding new capability to remember search query, for now it has been called only in Rich-Text-editor.  To support note related object I will apply new changes but I am not sure for the report page,  may be we should discuss with Maarteen.